axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/MrpServiceImpl.java

```diff
@@ -1504,29 +1504,86 @@ protected void copyMrpLineOrigins(MrpLine mrpLine, List<MrpLineOrigin> mrpLineOr
 
   @Override
   @Transactional(rollbackOn = {Exception.class})
-  public void generateProposals(Mrp mrp, boolean isProposalPerSupplier) throws AxelorException {
+  public void generateSelectedProposals(Mrp mrp, boolean isProposalPerSupplier)
+      throws AxelorException {
 
     Map<Pair<Partner, LocalDate>, PurchaseOrder> purchaseOrders = new HashMap<>();
     Map<Partner, PurchaseOrder> purchaseOrdersPerSupplier = new HashMap<>();
+    List<MrpLine> mrpLineList = getSelectedMrpLines(mrp);
+
+    generateProposals(
+        isProposalPerSupplier, purchaseOrders, purchaseOrdersPerSupplier, mrpLineList);
+  }
+
+  protected List<MrpLine> getSelectedMrpLines(Mrp mrp) throws AxelorException {
     List<MrpLine> mrpLineList =
         mrpLineRepository
             .all()
-            .filter("self.mrp.id = ?1 AND self.proposalToProcess = true", mrp.getId())
+            .filter(
+                "self.mrp.id = ?1 AND self.proposalToProcess = true AND self.proposalGenerated = false",
+                mrp.getId())
             .order("maturityDate")
             .fetch();
+    if (mrpLineList.isEmpty()) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_INCONSISTENCY,
+          I18n.get(SupplychainExceptionMessage.MRP_GENERATE_PROPOSAL_NO_LINE_SELECTED));
+    }
+    return mrpLineList;
+  }
 
-    for (MrpLine mrpLine : mrpLineList) {
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void generateAllProposals(Mrp mrp, boolean isProposalsPerSupplier) throws AxelorException {
+    Map<Pair<Partner, LocalDate>, PurchaseOrder> purchaseOrders = new HashMap<>();
+    Map<Partner, PurchaseOrder> purchaseOrdersPerSupplier = new HashMap<>();
+    List<MrpLine> mrpLineList = getAllMrpLines(mrp);
+    generateProposals(
+        isProposalsPerSupplier, purchaseOrders, purchaseOrdersPerSupplier, mrpLineList);
+  }
 
-      if (!mrpLine.getProposalGenerated()) {
-        mrpLineService.generateProposal(
-            mrpLine, purchaseOrders, purchaseOrdersPerSupplier, isProposalPerSupplier);
+  protected List<MrpLine> getAllMrpLines(Mrp mrp) throws AxelorException {
+    List<MrpLine> mrpLineList =
+        mrpLineRepository
+            .all()
+            .filter("self.mrp.id = :id AND self.proposalGenerated = false")
+            .bind("id", mrp.getId())
+            .bind("elementSelect", MrpLineTypeRepository.ELEMENT_PURCHASE_PROPOSAL)
```

Comment:
this bind is not used in the filter.
I'm not sure about this method. Most MRP have a lot of lines, here we fetch all lines and not only the proposals, which can cause huge performance issues. We should at least filter to get only proposal lines, maybe manage JPA cache.